addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoHandlerFactory.java

```diff
@@ -0,0 +1,171 @@
+     * @param discoveryServiceRegistry the non-null {@link NetworkAddressService} to use
+     */
+    @Reference
+    protected void setDiscoveryServiceRegistry(DiscoveryServiceRegistry discoveryServiceRegistry) {
+        Objects.requireNonNull(discoveryServiceRegistry, "discoveryServiceRegistry cannot be null");
+        this.discoveryServiceRegistry = discoveryServiceRegistry;
+    }
+
+    /**
+     * Unsets the {@link HttpService}
+     *
+     * @param discoveryServiceRegistry the {@link HttpService} (not used in this implementation)
+     */
+    protected void unsetDiscoveryServiceRegistry(DiscoveryServiceRegistry discoveryServiceRegistry) {
+        this.discoveryServiceRegistry = null;
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        Objects.requireNonNull(thingTypeUID, "thingTypeUID cannot be null");
+
+        return NeeoConstants.BINDING_ID.equals(thingTypeUID.getBindingId());
+    }
+
+    @Nullable
+    @Override
+    protected ThingHandler createHandler(Thing thing) {
+        Objects.requireNonNull(thing, "thing cannot be null");
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(BRIDGE_TYPE_BRAIN)) {
+            final HttpService localHttpService = httpService;
+            final NetworkAddressService localNetworkAddressService = networkAddressService;
+
+            Objects.requireNonNull(localHttpService, "HttpService cannot be null");
+            Objects.requireNonNull(localNetworkAddressService, "networkAddressService cannot be null");
+
+            final int port = HttpServiceUtil.getHttpServicePort(this.bundleContext);
+
+            startScan(NeeoConstants.SCAN_ROOMS); // starts scan for rooms
+            return new NeeoBrainHandler((Bridge) thing, port < 0 ? NeeoConstants.DEFAULT_BRAIN_HTTP_PORT : port,
+                    localHttpService, localNetworkAddressService);
+        } else if (thingTypeUID.getId().startsWith("room")) {
+            startScan(NeeoConstants.SCAN_DEVICES); // starts scan for devices
+            return new NeeoRoomHandler((Bridge) thing);
+        } else if (thingTypeUID.getId().startsWith("device")) {
+            return new NeeoDeviceHandler(thing);
+        }
+
+        return null;
+    }
+
+    /**
+     * Helper method to start a new discovery scan for NEEO after a specified number of milliseconds (gives time for the
+     * system to create the thing properly).
+     *
+     * @param waitMilliSeconds a >=0 milliseconds to schedule the scan
+     */
+    private void startScan(int waitMilliSeconds) {
+        if (waitMilliSeconds < 0) {
+            throw new IllegalArgumentException("waitMilliSeconds cannot be below 0: " + waitMilliSeconds);
+        }
+        final DiscoveryServiceRegistry localRegistry = discoveryServiceRegistry;
+
+        if (localRegistry != null) {
+            scheduler.schedule(() -> {
+                localRegistry.startScan(NeeoConstants.BINDING_ID, null);
```

Comment:
please remove the registry and replace it with you own `DiscoveryService` which you can start here, similar to https://github.com/eclipse/smarthome/blob/bc9d6284c54ca38b13436abf733d567b1e3c5d6d/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HueThingHandlerFactory.java#L104-L109